### DISCUSSION TITLE
Toast notification on windows

### DISCUSSION THREAD

**Discussion by DennisWelu** [CommunityToolkit/Maui#1178] [General]
I just evaluated the Toast feature in the toolkit. Kudos to the contributing developers. But I am curious about the decision to use a windows toast notification on the Windows platform. 1) The rendering "toast" on Windows is notably different / more formal, 2) it's outside of the application visual container, and 3) it registers something in the system (toast remains in the notification list for Windows until dismissed). Toast for Android and iOS feel considerably more lightweight from a user perspective, being internal to the app only, temporary in the app container, then gone. The use case scenarios you would consider for the Toast UI become limited if you wish to include Windows support.

Is there any consideration for changing this behavior on Windows to use something more similar to the iOS/Android experience?

**Comment by TheCodeTraveler** [comment]
@VladislavAntonyuk Would the upcoming Popup PR also address this 👆? 

https://github.com/CommunityToolkit/Maui/pull/1086

**Reply by VladislavAntonyuk** [reply]
No.
Popup and toast have different implementation. I would suggest keeping them separate.

**Comment by VladislavAntonyuk** [comment]
It is closer to "native" implementation and more familiar for users to see something in Windows notifications. Also toast should be automatically dismissed (at least we set an expiration time for it).
In Xamarin community Toolkit we injected toast in Application container template (grid). It had some downsides (main content size was changed depending on toast size)
In MAUI the template was changed.

I saw a sample in Windows community Toolkit that looks like on Android and iOS. You can try reuse it in your app with custom IToast implementation.

**Comment by DennisWelu** [comment]
@VladislavAntonyuk Good thought about using the IToast interface for a custom approach.

I appreciate the "closer to native" implementation approach, but in this case it seems like it's not exactly the same kind of native "thing" despite being called a ToastNotification in WinUI. It really behaves like something different. Some similarities sure, but different enough that if you are thinking about cross-platform user experience it stands out as "not the same kind of thing" on windows. That's my 2 cents anyway. I'll check out the IToast approach or just do something custom altogether. I understand how the root grid approach you mentioned can impact other content, but maybe using a standard page ControlTemplate to provide a layout that hosts a toast message consistently across pages would suffice.

**Reply by VladislavAntonyuk** [reply]
You can either have the exact same look and feel on all platforms with a custom implementation, or you can use different "native" features of the platforms. I prefer the second approach for better OS integration

**Reply by DennisWelu** [reply]
I "tend" to feel the same...just feel in this case the windows implementation is too far different. But no worries! There are other ways.

**Comment by CliffAgius** [comment]
Closed as per the Community Stand-up discussion here: https://youtu.be/N9wMcBP4jtg?t=2889

**Reply by DennisWelu** [reply]
@CliffAgius Can you write a sentence summary of the conclusion from the discussion? (The video is an hour long)
